src/test/java/org/javarosa/core/test/Scenario.java

```diff
@@ -180,6 +180,17 @@ public ValidateOutcome getValidationOutcome() {
      * For this reason, this method will try to detect these predicates,
      * turn them into multiplicity values, and remove them from the output
      * reference.
+     *
+     * When using the result of this method for test assertions against {@link FormIndex#getReference()} we need to
+     * specify multiplicity on all steps. For example the following would pass for a form index pointing at a
+     * question at the top level of the form:
+     *
+     * <code>
+     * assertThat(formIndex.getReference(), is(getRef("/data/question[0]")));
+     * </code>
+     *
+     * If we used just used <code>getRef("/data/question")</code> the test would fail as the multiplicity defaults
+     * to <code>0</code>.
```

Comment:
That makes it sound like it should pass. I think what you mean is that when using `getRef`, the multiplicity for `question` would be unbound (-1) but that for any reference used "for real" the multiplicity would have defaulted to 0. 

Maybe it's worth it to explain that `getRef` has no way of knowing whether a node without a multiplicity is an unbounded repeat or a singular group or question so it assumes the former. Adding an explicit `[0]` forces it to build a reference for the later. I had to go through and figure that out again just now.

Reply:
> Maybe it's worth it to explain that getRef has no way of knowing whether a node without a multiplicity is an unbounded repeat or a singular group or question so it assumes the former. Adding an explicit [0] forces it to build a reference for the later. I had to go through and figure that out again just now.

This is the best explanation we've had so far so I'm going to try and use it!